types/typescript.d.ts

```diff
@@ -213,7 +213,6 @@ interface Events {
 }
 
 interface Props extends Events {
-	[props: string]: any
```

Comment:
This would mean we can't pass unknown properties as props. Is this what we want?

Reply:
@thysultan Correct. 

Consider the following:

hello.tsx
```tsx
import { PureComponent } from "dio.js"

interface Props {
  username?: string
}

export default class Hello extends PureComponent<Props> {
  render () {
    return <div>Hello, {this.props.username}</div>
  }
}
```

index.tsx
```tsx
import * as dio from "dio.js"
import Hello from "./hello"

dio.render(
  <Hello usermame="john" asdf={123} />,
  document.getElementById("app")
)

```

Under current circumstances, this will typecheck, allowing a typo and an unused prop to be passed in.
With unknown properties disallowed (this is React's approach as well), there is much less room for errors and more benefits of static typing to be reaped.

Reply:
Makes sense. In that case would we also need to change 367: `interface IntrinsicElements {[props: string]: any}`

Reply:
@thysultan Yes, that would be great, but it is a non-trivial change as one would need to enumerate all the possible html tags. Look for `InstrinsicElements` at https://unpkg.com/@types/react@16.0.34/index.d.ts to get an idea.

Much rather this on separately with a follow-up PR